src/components/content/monitor/Monitor.tsx

```diff
@@ -0,0 +1,282 @@
+            if (rsp.length > 0) {
+                rsp.forEach((serviceVo: ServiceVo) => {
+                    const cusServiceName: { value: string; label: string } = {
+                        value: serviceVo.customerServiceName ?? '',
+                        label: serviceVo.customerServiceName ?? '',
+                    };
+                    cusServiceNameList.push(cusServiceName);
+                    const serviceName: { value: string; label: string } = {
+                        value: serviceVo.name,
+                        label: serviceVo.name,
+                    };
+
+                    serviceNameList.push(serviceName);
+                });
+                setData(rsp);
+                setServiceNameList(serviceNameList);
+                setCustomerServiceNameList(cusServiceNameList);
+            }
+        });
+    }, []);
+
+    const handleChangeServiceName = (selectServiceName: string) => {
+        if (selectServiceName) {
+            const cusServiceNameList: { value: string; label: string }[] = [];
+            data.forEach((serviceVo: ServiceVo) => {
+                if (serviceVo.name === selectServiceName) {
+                    const cusServiceName: { value: string; label: string } = {
+                        value: serviceVo.customerServiceName ?? '',
+                        label: serviceVo.customerServiceName ?? '',
+                    };
+                    cusServiceNameList.push(cusServiceName);
+                }
+            });
+            setCustomerServiceNameList(cusServiceNameList);
+        }
+    };
+
+    const handleChangeCustomerServiceName = (selectCustomerServiceName: string) => {
+        form.setFieldsValue({ ServiceId: getServiceId(selectCustomerServiceName) });
+    };
+
+    const getServiceId = (selectCustomerServiceName: string): string => {
+        let currentID: string = '';
+        if (selectCustomerServiceName) {
+            data.forEach((serviceVo) => {
+                if (selectCustomerServiceName === serviceVo.customerServiceName) {
+                    currentID = serviceVo.id;
+                }
+            });
+        }
+        return currentID;
+    };
+
+    const layout = {
+        labelCol: { span: 8 },
+        wrapperCol: { span: 16 },
+    };
+
+    const tailLayout = {
+        wrapperCol: { offset: 8, span: 16 },
+    };
+
+    const onFinish = (values: { ServiceName: string; CustomerServiceName: string; ServiceId: string }) => {
+        queue.clear();
+        if (timer) {
+            clearInterval(timer);
+        }
+        if (!values.ServiceId) {
+            return;
+        }
+        const selectedServiceId = values.ServiceId;
+        setServiceId(selectedServiceId);
+        void MonitorService.getMetrics(selectedServiceId).then((rsp: Metric[]) => {
+            if (rsp.length > 0) {
+                if (queue.isFull()) {
+                    queue.dequeue();
+                }
+                queue.enqueue(rsp);
```

Comment:
dequeue only removes the first element from the queue. 
where as enqueue here can have multiple values. 
how is then this handled when the new response length is more than 1?

Reply:
fixed.